### PR TITLE
boards: cc3220sf: Fix compatibility issue with gnuarmemb

### PR DESCRIPTION
**Pull request by SebastianBoe**
The cc3220sf_launchxl board is AFAICT not compatible with the
toolchain variant 'gnuarmemb'. To avoid triggering CI failures we
remove this compatibility declaration from the metadata.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

**Comment by SebastianBoe** [comment]
NB: I am certain that cc3220sf_launchxl is broken with gnuarmemb. Try the sample samples/net/wifi to observe this.

But I don't know why, or if this is expected. So it might be that the correct fix here is to resolve the incompatibility, not to drop compatibility.

**Comment by SebastianBoe** [comment]
Error message:

```
ccache /home/sebo/Downloads/gcc-arm-none-eabi-7-2018-q2-update/bin/arm-none-eabi-gcc -DBUILD_VERSION=zephyr-v1.13.0-928-g9e676b485f -DKERNEL -DSL_PLATFORM_MULTI_THREADED -DSL_SUPPORT_IPV6 -DUSE_CC3220_ROM_DRV_API -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I../../../../kernel/include -I../../../../arch/arm/include -I../../../../soc/arm/ti_simplelink/cc32xx -I../../../../soc/arm/ti_simplelink/cc32xx/include -I../../../../soc/arm/ti_simplelink/include -I../../../../boards/arm/cc3220sf_launchxl -I../../../../include -I../../../../include/drivers -Izephyr/include/generated -isystem /home/sebo/Downloads/gcc-arm-none-eabi-7-2018-q2-update/bin/../lib/gcc/arm-none-eabi/7.3.1/include -isystem /home/sebo/Downloads/gcc-arm-none-eabi-7-2018-q2-update/bin/../lib/gcc/arm-none-eabi/7.3.1/include-fixed -I/home/sebo/Downloads/gcc-arm-none-eabi-7-2018-q2-update/arm-none-eabi/include -I../../../../ext/hal/cmsis/Include -I../../../../ext/hal/ti/simplelink/source/ti/devices/cc32xx/. -I../../../../ext/hal/ti/simplelink/source/ti/devices/cc32xx/inc -I../../../../ext/hal/ti/simplelink/source/ti/devices/cc32xx/driverlib -I../../../../ext/hal/ti/simplelink/. -I../../../../ext/hal/ti/simplelink/source -I../../../../ext/hal/ti/simplelink/kernel/zephyr/dpl -I../../../../drivers/wifi/simplelink/. -I../../../../ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting -I../../../../ext/hal/ti/simplelink/source/ti/drivers/net/wifi/bsd -Og -g -Wall -Wformat -Wformat-security -Wno-format-zero-length -imacros /home/sebo/zephyr/samples/net/wifi/b/zephyr/include/generated/autoconf.h -ffreestanding -Wno-main -mthumb -mcpu=cortex-m4 -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -Wno-pointer-sign -Wno-unused-but-set-variable -fno-reorder-functions -fno-defer-pop -Werror=implicit-int -Wpointer-arith -ffunction-sections -fdata-sections -mabi=aapcs -march=armv7e-m -std=c99 -MD -MT zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj   -c /home/sebo/zephyr/arch/arm/core/offsets/offsets.c
In file included from ../../../../kernel/include/kernel_structs.h:10:0,
                 from /home/sebo/zephyr/arch/arm/core/offsets/offsets.c:26:
../../../../include/kernel.h: In function 'k_enable_sys_clock_always_on':
../../../../include/kernel.h:1677:10: error: 'ENOTSUP' undeclared (first use in this function)
  return -ENOTSUP;
          ^~~~~~~
../../../../include/kernel.h:1677:10: note: each undeclared identifier is reported only once for each function it appears in

```

**Comment by GAnthony** [comment]
LGTM.
The 'gnuarmemb' was a copy-paste error.  I've only been building with Zephyr toolchain up to now.
I know there is more interest in adding the TI CGT, but haven't had the cycles to get back to that effort.

**Review by GAnthony — APPROVED** [review]
LGTM

**Comment by galak** [comment]
> Error message:

what test shows this?

**Comment by galak** [comment]
> NB: I am certain that cc3220sf_launchxl is broken with gnuarmemb. Try the sample samples/net/wifi to observe this.

The issue I don't think is exactly cc3220sf_launchxl related, its that we are pickup errno.h from ./ext/hal/ti/simplelink/source/ti/drivers/net/wifi/bsd/errno.h when building with gnuarmemb, not sure why right now.

**Comment by galak** [comment]
Ok, so the issue is related to how we set the system includes for zephyr toolchain vs gnuarmemb. For zephyr we set --sysroot /opt/zephyr-sdk-0.9.3/sysroots/armv5-zephyr-eabi/usr (which gets errno.h from ther).  On gnuarmemb we don't set --sysroot

**Comment by galak** [comment]
@GAnthony any idea what's the implication if we drop include the bsd dir:

```
diff --git a/drivers/wifi/simplelink/CMakeLists.txt b/drivers/wifi/simplelink/CMakeLists.txt
index 43fc9efb41..bfbf59beb8 100644
--- a/drivers/wifi/simplelink/CMakeLists.txt
+++ b/drivers/wifi/simplelink/CMakeLists.txt
@@ -4,7 +4,6 @@ if(CONFIG_WIFI_SIMPLELINK)
     $ENV{ZEPHYR_BASE}/ext/hal/ti/simplelink/kernel/zephyr/dpl
     $ENV{ZEPHYR_BASE}/ext/hal/ti/simplelink/source
     $ENV{ZEPHYR_BASE}/ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting
-    $ENV{ZEPHYR_BASE}/ext/hal/ti/simplelink/source/ti/drivers/net/wifi/bsd
     )
   zephyr_sources(
     simplelink_support.c
```

**Comment by GAnthony** [comment]
@galak, I think the bsd include dir line can be removed (it compiles without it there).
I think that bsd include directory is a vestige from older days when the simplelink driver was calling the TI BSD socket APIs.   Now, it's calling underlying sl_ socket APIs.
The SDK host driver needs both the system errno.h and SDK's errno.h to do a mapping of error numbers, but that is handled by explicitly including <simplelink>.../bsd/errno.h in the source.
So, bottom line, I think it is OK to remove.

**Comment by SebastianBoe** [comment]
Updated the PR to remove the include path instead of dropping support for gnuarmemb.

**Review by galak — APPROVED** [review]
